### PR TITLE
Start moving to spock2.0

### PR DESCRIPTION
**Pull request by rtretyak**
- add dependencies
- remove spring context ext
- remove rerun ext
- remove setup once ext
- update assumptions
- update shared+autowired fields

### CODE REVIEW

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/TestCaseReportCreator.groovy [hunk 1]
```diff
@@ -1,48 +1,48 @@
...
-        //create table of contents at the top of the report
-        def toc = specs.collect {
-            def name = Utils.getSpecClassName(it).split("\\.").last() - ~/Spec$/
-            "- [$name](#${name.toLowerCase()})\n"
-        }.join("")
-        summaryFile.text = toc + summaryFile.text
-
-    }
-}
+//package org.openkilda.functionaltests.helpers
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/TestCaseReportCreator.groovy:1]: how about adding some comments at the beginning of the file?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy [hunk 2]
```diff
@@ -225,118 +225,146 @@ class SwapEndpointSpec extends HealthCheckSpecification {
     @Tidy
     @Unroll
     @IterationTag(tags = [LOW_PRIORITY], iterationNameRegex = /src1/)
-    def "Able to swap #endpointsPart (#description) for two flows with the same source and different destination \
+    def "Able to swap #data.endpointsPart (src1 <-> dst2, dst1 <-> src2) for two flows with the same source and different destination \
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy:228]: can we return the `description` variabe into the `where` block and description of the test?
if no, we need to delete `@IterationTag(tags = [LOW_PRIORITY], iterationNameRegex = /src1/)`. it is not actual in this case.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy [hunk 3]
```diff
@@ -549,24 +579,24 @@ switches"() {
...
         flowHelper.deleteFlow(flow1.id)
-        !isTestComplete && [switchPair.src.dpId, switchPair.dst.dpId].each { northbound.synchronizeSwitch(it, true)}
+        !isTestComplete && [switchPair.src.dpId, switchPair.dst.dpId].each { northbound.synchronizeSwitch(it, true) }
     }
 
     @Tidy
     @Unroll
     @Tags(LOW_PRIORITY)
-    def "Unable to swap #endpointsPart for two flows (#description)"() {
+    def "Unable to swap #data.endpointsPart for two flows (#data.description)x"() {
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy:588]: `flows (#data.description)x` ? (`x` at the end)

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/rerun/RerunExtension.groovy [hunk 4]
```diff
@@ -4,6 +4,7 @@ import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension
 import org.spockframework.runtime.extension.IMethodInterceptor
 import org.spockframework.runtime.extension.IMethodInvocation
 import org.spockframework.runtime.model.FeatureInfo
+import spock.lang.Specification
```
Comment (andriidovhan) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/rerun/RerunExtension.groovy:7]: Why do we need this import?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/unit/spec/WrappersSpec.groovy [hunk 5]
```diff
@@ -27,6 +27,6 @@ class WrappersSpec extends Specification {
         Wrappers.wait(timeout, condition)
 
         where:
-        condition << [{ true }, { assert true }, { }, { new Object() }]
+        condition << [{ true }, /*{ assert true },*/ { }, { new Object() }]
```
[the remark this answers is not in the document]
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/unit/spec/WrappersSpec.groovy]: Was due to https://github.com/spockframework/spock/issues/1287, which is now fixed. uncommented

File: src-java/testing/functional-tests/src/test/resources/META-INF/services/org.spockframework.runtime.extension.IGlobalExtension [hunk 6]
```diff
@@ -1,9 +1,9 @@
-org.openkilda.functionaltests.extension.spring.SpringContextExtension
+#org.openkilda.functionaltests.extension.spring.SpringContextNotifier
 org.openkilda.functionaltests.extension.SkippedTestsLogger
 org.openkilda.functionaltests.extension.fixture.SetupOnceExtension
 org.openkilda.functionaltests.extension.tags.TagExtension
 org.openkilda.functionaltests.extension.env.AssumeProfileExtension
-org.openkilda.functionaltests.extension.env.HardwareEnvCleanupExtension
+#org.openkilda.functionaltests.extension.env.HardwareEnvCleanupExtension
```
[the remark this answers is not in the document]
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/resources/META-INF/services/org.spockframework.runtime.extension.IGlobalExtension]: ye, thx

**Comment by rtretyak** [comment]
All review findings fixed in a separate commit

**Pull request merged**